### PR TITLE
Sphinx docs

### PR DESCRIPTION
**Pull request by funkyHat**

### CODE REVIEW

**Comment by funkyHat** [comment]
The test has finished, circleci is being slow today.

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+~~~~~~~~
+
+.. highlight:: bash
+
+* You will need SSH keys generated for both the manager and the application VM's. If you are using the default key locations in the inputs, these can be created with the following commands::
+
+    ssh-keygen -b2048 -N "" -q -f ~/.ssh/cloudify-manager-kp.pem
+    ssh-keygen -b2048 -N "" -q -f ~/.ssh/cloudify-agent-kp.pem
+
+* A working vSphere environment. The plugin was tested with version 5.5.
```
Comment (geokala) [inline docs/index.rst]: Can you change this to 6.0

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
+# -*- coding: utf-8 -*-
```
Comment (geokala) [inline docs/conf.py]: Just to note that I discussed this with @funkyHat and this boilerplate is fairly standard for sphinxy stuff and very non-trivial to move to a separate repo.

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  * On the datacenter:
+
+    * dvPort group/Create
+    * dvPort group/Delete
+
+
+OS Templates
+~~~~~~~~~~~~
+
+* You need two OS templates of your preferred operating systems (e.g. Ubuntu Trusty) within the vSphere datastores. One for the Cloudify manager and one for the application VMs. The application VM template should accept the Cloudify agent public key for its root user. The Cloudify manager template must accept the cloudify manager public key. Note that you can choose to use same template for both the manager and the application VMs, in that case the shared template must accept both public keys.
```
Comment (geokala) [inline docs/index.rst]: Can we change the e.g. to Centos 7? (Ubuntu won't work for the manager any more)

File: docs/index.rst [hunk 4]
```diff
@@ -83,7 +83,13 @@ Permissions on vCenter
 OS Templates
 ~~~~~~~~~~~~
 
-* You need two OS templates of your preferred operating systems (e.g. Ubuntu Trusty) within the vSphere datastores. One for the Cloudify manager and one for the application VMs. The application VM template should accept the Cloudify agent public key for its root user. The Cloudify manager template must accept the cloudify manager public key. Note that you can choose to use same template for both the manager and the application VMs, in that case the shared template must accept both public keys.
+* You need two OS templates within the vSphere datastores.
+  One for the Cloudify manager and one for the application VMs.
+  The Cloudify manager template must have CentOS 7 installed.
+  The application VM template should accept the Cloudify agent public key for its root user.
+  The Cloudify manager template must accept the cloudify manager public key.
+  Note that you can choose to use same template for both the manager and the application VMs,
```
Comment (chanagershom) [inline docs/index.rst]: `Note that you can choose to use same template for both ~the manager~ *Cloudify Manager* and the application VMs`...

File: docs/index.rst [hunk 5]
```diff
@@ -83,7 +83,13 @@ Permissions on vCenter
 OS Templates
 ~~~~~~~~~~~~
 
-* You need two OS templates of your preferred operating systems (e.g. Ubuntu Trusty) within the vSphere datastores. One for the Cloudify manager and one for the application VMs. The application VM template should accept the Cloudify agent public key for its root user. The Cloudify manager template must accept the cloudify manager public key. Note that you can choose to use same template for both the manager and the application VMs, in that case the shared template must accept both public keys.
+* You need two OS templates within the vSphere datastores.
+  One for the Cloudify manager and one for the application VMs.
```
Comment (chanagershom) [inline docs/index.rst]: `You need two OS templates within the vSphere datastores, one for the Cloudify Manager and one for the application VMs.`

Also, `_Cloudify Manager_ needs to be capitalized everywhere. It is a product name, so even though this is something small, it is important it is used correctly.

**Pull request merged**